src/pipecat/services/hathora/tts.py

```diff
@@ -0,0 +1,229 @@
+#
+# Copyright (c) 2024–2025, Daily
+#
+# SPDX-License-Identifier: BSD 2-Clause License
+#
+
+"""[Hathora-hosted](https://models.hathora.dev) text-to-speech services."""
+
+import io
+import os
+import wave
+from typing import Optional, Tuple
+
+import aiohttp
+from loguru import logger
+
+from pipecat.frames.frames import (
+    ErrorFrame,
+    TTSAudioRawFrame,
+    TTSStartedFrame,
+    TTSStoppedFrame,
+)
+from pipecat.services.tts_service import TTSService
+
+def _decode_audio_payload(
+    audio_bytes: bytes,
+    *,
+    fallback_sample_rate: int,
+    fallback_channels: int = 1,
+) -> Tuple[bytes, int, int]:
+    """Convert a WAV/PCM payload into raw PCM samples for TTSAudioRawFrame."""
+
+    try:
+        with wave.open(io.BytesIO(audio_bytes), "rb") as wav_reader:
+            channels = wav_reader.getnchannels()
+            sample_rate = wav_reader.getframerate()
+            frames = wav_reader.readframes(wav_reader.getnframes())
+            return frames, sample_rate, channels
+    except (wave.Error, EOFError):
+        # If the payload is already raw PCM, just pass it through.
+        return audio_bytes, fallback_sample_rate, fallback_channels
+
+class KokoroTTSService(TTSService):
+    """Kokoro is an open-weight TTS model with 82 million parameters.
+
+    This service uses the Hathora-hosted Kokoro model via the HTTP API.
+
+    [Documentation](https://models.hathora.dev/model/hexgrad-kokoro-82m)
+    """
+
+    def __init__(
+        self,
+        *,
+        base_url = None,
+        api_key = None,
+        voice: Optional[str] = None,
+        speed: Optional[float] = None,
+        **kwargs,
+    ):
+        """Initialize the Hathora-hosted Kokoro TTS service.
+
+        Args:
+            base_url: Base URL for the Hathora Kokoro TTS API, .
+            api_key: API key for authentication with the Hathora service;
+                provisiion one [here](https://models.hathora.dev/tokens).
+            voice: Voice to use for synthesis (see the
+                [Hathora docs](https://models.hathora.dev/model/hexgrad-kokoro-82m)
+                for the default value; [list of voices](https://huggingface.co/hexgrad/Kokoro-82M/blob/main/VOICES.md)).
+            speed: Speech speed multiplier (0.5 = half speed, 2.0 = double speed, default: 1).
+        """
+        super().__init__(
+            **kwargs,
+        )
+        self._base_url = base_url
+        self._api_key = api_key
+        self._voice = voice
+        self._speed = speed
+
+    def can_generate_metrics(self) -> bool:
+        return True
+
+    async def run_tts(self, text: str):
```

Comment:
Same as above, re: docstring.

Reply:
Also, you will want to add tracing to this function, which can be done by decorating the `run_tts()` function:
```suggestion
    @traced_tts
    async def run_tts(self, text: str):
```

Import is:
```
from pipecat.utils.tracing.service_decorators import traced_tts
```